calour/manipulation.py

```diff
@@ -66,7 +66,7 @@ def join_fields(exp, field1, field2, newname=None, axis=0, separator='_', inplac
         raise ValueError('new field name %s alreay in metadata. Please use different newname value' % newname)
 
     # add the new column
-    newcol = metadata[field1].str.cat(metadata[field2].astype(str), sep=separator)
+    newcol = metadata[field1].astype(str).str.cat(metadata[field2].astype(str), sep=separator)
```

Comment:
why not just use list comphrehesion - it is easier to understand and easier to see if there is error.

`newcol = [str(i) + sparator + str(j) for i, j in zip(metadata[field1], metadata[field2])]`

BTW, in line 52, similarly delete ` raise ValueError('incorrect axis %s. please use 0/1' % axis)`.